### ISSUE TITLE
No longer make copyright lines stringly typed

### ISSUE THREAD

**Issue by carmenbianca** [fsfe/reuse-tool#947]
Presently, copyright lines are stringly typed in the tool. The whole thing `SPDX-FileCopyrightText: 2016, 2018--2020, 2023--present Carmen Bianca BAKKER <carmenbianca@fsfe.org>` is one string.

I think it might make sense to write some kind of `CopyrightLine` class that can parse elements from such strings. For example, the above properties of the above example:

- copyright_prefix - `SPDX-FileCopyrightText`
- date_range - some DateRange object that can encode the concept of `2016`, `2018--2020`, and `2023--present`. Or, if that all is too over-engineered, just the string `2016, 2018--2020, 2023--present`
- name - `Carmen Bianca BAKKER`
- contact - `carmenbianca@fsfe.org`

If implemented well, this could heavily simplify date-range merging, which is currently spaghetti code.

**Comment by silverhook** [comment]
I like this.

Two changes I’d like to propose though:
- make contact (or at least the usage of `<` and `>`) optional
- include a `©` (or `(c)` or `(C)`) sign at the beginning, in order to make sure that the copyright notice actually complies with how it is defined in pretty much all jurisdictions (I [wrote about this](https://matija.suklje.name/how-and-why-to-properly-write-copyright-statements-in-your-code#why-the-sign) some time ago)

**Comment by silverhook** [comment]
Personally I’d limit it down to:

```
    spdx-symbol:    SPDX-FileCopyrightText: © <year> <statement>
    spdx-c:         SPDX-FileCopyrightText: (c) <year> <statement>
    spdx-C:         SPDX-FileCopyrightText: (C) <year> <statement>
```

Maaaybeee include:
```
    spdx:           SPDX-FileCopyrightText: <year> <statement>
```

…but you know my thoughts on this ;)

**Comment by 83123rn** [comment]
Yes, I can see how many mistakes I made.  Maybe it would be better to start again, but this time. He doesn't rush things. , sorry people for the inconvenience

**Comment by 83123rn** [comment]
I read quite a lot of things, but if I make a new one, don't be too angry?  This became so catastrophic because I didn't think it through, and I was rushing to do it.  Now I read everything thoroughly, but sometimes it's a bit too much 🙂  I'm sorry in advance from everyone
